### ISSUE TITLE
src/rendering/semantics.dart: Failed assertion

### ISSUE THREAD

**Issue by benwaffle** [flutter/flutter#10069]
Code:
```dart
import 'package:flutter/material.dart';

void main() {
  runApp(new MaterialApp(
      title: 'Flutter',
      theme: new ThemeData(primarySwatch: Colors.red),
      home: new MyHomePage()));
}

class MyHomePage extends StatefulWidget {
  @override
  _MyHomePageState createState() => new _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  String apikey;

  @override
  Widget build(BuildContext context) {
    return new Scaffold(
        appBar: new AppBar(
          title: new Text('Flutter'),
        ),
        body: new Center(
            child: new Container(
                padding: new EdgeInsets.all(20.0),
                child: new Column(children: [
                  new TextField(
                      decoration:
                          new InputDecoration(labelText: 'Username')),
                  new TextField(
                      obscureText: true,
                      decoration: new InputDecoration(labelText: 'Password')),
                  new FlatButton(
                    onPressed: () {
                      var dialog = new SimpleDialog(title: new Text('hello'));
                      showDialog(context: context, child: dialog);
                    },
                    child: new Text('Login'),
                  )
                ]))));
  }
}
```

Error:
```
I/flutter (17028): ══╡ EXCEPTION CAUGHT BY RENDERING LIBRARY ╞═════════════════════════════════════════════════════════
I/flutter (17028): The following assertion was thrown during _updateSemantics():
I/flutter (17028): 'package:flutter/src/rendering/semantics.dart': Failed assertion: line 339 pos 16:
I/flutter (17028): 'seenChildren.add(child)': is not true.
I/flutter (17028): 
I/flutter (17028): Either the assertion indicates an error in the framework itself, or we should provide substantially
I/flutter (17028): more information in this error message to help you determine and fix the underlying cause.
I/flutter (17028): In either case, please report this assertion by filing a bug on GitHub:
I/flutter (17028):   https://github.com/flutter/flutter/issues/new
I/flutter (17028): 
I/flutter (17028): When the exception was thrown, this was the stack:
I/flutter (17028): #2      SemanticsNode.addChildren.<anonymous closure> (package:flutter/src/rendering/semantics.dart:339:16)
I/flutter (17028): #4      SemanticsNode.addChildren (package:flutter/src/rendering/semantics.dart:336:12)
I/flutter (17028): #5      _InterestingSemanticsFragment.compile (package:flutter/src/rendering/object.dart:737:12)
I/flutter (17028): #6      _SyncIterator.moveNext (dart:core-patch/core_patch.dart:62)
I/flutter (17028): #7      List.addAll (dart:core-patch/growable_array.dart:197)
I/flutter (17028): #8      SemanticsNode.addChildren (package:flutter/src/rendering/semantics.dart:322:18)
I/flutter (17028): #9      _InterestingSemanticsFragment.compile (package:flutter/src/rendering/object.dart:737:12)
I/flutter (17028): #10     _SyncIterator.moveNext (dart:core-patch/core_patch.dart:62)
I/flutter (17028): #11     Iterable.single (dart:core/iterable.dart:501)
I/flutter (17028): #12     RenderObject._updateSemantics (package:flutter/src/rendering/object.dart:2495:88)
I/flutter (17028): #13     PipelineOwner.flushSemantics (package:flutter/src/rendering/object.dart:1200:16)
I/flutter (17028): #14     BindingBase&SchedulerBinding&GestureBinding&ServicesBinding&RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:236:19)
I/flutter (17028): #15     BindingBase&SchedulerBinding&GestureBinding&ServicesBinding&RendererBinding&WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:361:22)
I/flutter (17028): #16     BindingBase&SchedulerBinding&GestureBinding&ServicesBinding&RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:170:5)
I/flutter (17028): #17     BindingBase&SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:685:15)
I/flutter (17028): #18     BindingBase&SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:633:9)
I/flutter (17028): #19     BindingBase&SchedulerBinding&GestureBinding&ServicesBinding&RendererBinding.scheduleWarmUpFrame.<anonymous closure> (package:flutter/src/rendering/binding.dart:247:20)
I/flutter (17028): #21     _Timer._runTimers (dart:isolate-patch/timer_impl.dart:366)
I/flutter (17028): #22     _Timer._handleMessage (dart:isolate-patch/timer_impl.dart:394)
I/flutter (17028): #23     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:151)
I/flutter (17028): (elided 4 frames from class _AssertionError and package dart:async-patch)
Reloaded 1 of 425 libraries in 1,039ms.
I/flutter (17028): 
I/flutter (17028): The following RenderObject was being processed when the exception was fired:
I/flutter (17028):   RenderView#668411019
I/flutter (17028):   debug mode enabled - android
I/flutter (17028):   window size: Size(1440.0, 2392.0) (in physical pixels)
I/flutter (17028):   device pixel ratio: 3.5 (physical pixels per logical pixel)
I/flutter (17028):   configuration: Size(411.4, 683.4) at 3.5x (in logical pixels)
I/flutter (17028):   semantics enabled
I/flutter (17028): This RenderObject had the following descendants (showing up to depth 5):
I/flutter (17028):   RenderCustomPaint#324322237
I/flutter (17028):     RenderPointerListener#734188547
I/flutter (17028):       RenderAbsorbPointer#1018078217
I/flutter (17028):         RenderSemanticsAnnotations#1033215741
I/flutter (17028):           _RenderTheatre#482752555
I/flutter (17028): ════════════════════════════════════════════════════════════════════════════════════════════════════
```

**Comment by benwaffle** [comment]
flutter doctor is all checkmarks

```
[✓] Flutter (on Linux, channel alpha)
    • Flutter at /home/ben/dev/flutter
    • Framework revision dc4ca8db83 (2 days ago), 2017-05-11 15:33:29 -0700
    • Engine revision f99d3ef108
    • Tools Dart version 1.23.0-dev.11.11
```

**Comment by goderbauer** [comment]
I tried to reproduce this with the example code given and I didn't see an assertion. This might have been fixed with the work that went into accessibility recently. Please feel free to re-open this bug if you still see this assert (ideally with additional information explaining how to reproduce it).
